modal/_utils/git_utils.py

```diff
@@ -0,0 +1,80 @@
+# Copyright Modal Labs 2025
+import os
+import subprocess
+from typing import Dict, List, Optional, Union
+
+
+def run_command(args: List[str]) -> Optional[str]:
+    """Run a command and return its output.
+
+    Args:
+        args: Command and arguments as a list
+
+    Returns:
+        Command output as string or None if command failed
+    """
+    try:
+        result = subprocess.run(
```

Comment:
Small thing, since Python's asynchronous executor is single-threaded we try to avoid running [synchronous code inside an asynchronous function](https://docs.python.org/3/library/asyncio-task.html#asyncio.to_thread). Starting and waiting for a `git` subprocess takes about ~20-200 milliseconds and would halt execution of concurrent / background tasks in the rest of the Modal client while it happens.

Can stay fully async with a quick API change to [create_subprocess_exec()](https://docs.python.org/3/library/asyncio-subprocess.html#asyncio.create_subprocess_exec)

Reply:
Nice I didn't know there were async tools in subprocess. Could probably use that to parallelize some of the information gathering too?

Reply:
Thanks!!